frsca.cue

```diff
@@ -98,3 +120,59 @@ frsca: clusterPolicy?: [_]: {
         }]
     }]
 }
+
+// watch is used to add a tekton trigger to run a pipeline
+frsca: trigger?: [Name=_]: {
+	pipelineRun: pipelineV1Beta1.#PipelineRun & {
+		apiVersion: "tekton.dev/v1beta1"
+		kind:       "PipelineRun"
+		metadata: {
+			generateName: *"\(Name)-" | string
+			labels: "app.kubernetes.io/description": "PipelineRun"
+		}
+	}
+	let pr=pipelineRun
+	triggerTemplate: triggersV1Beta1.#TriggerTemplate & {
+		apiVersion: "triggers.tekton.dev/v1beta1"
+		kind: "TriggerTemplate"
+		metadata:	name: "\(Name)-triggertemplate"
+		spec: {
+			params: [{
+				name: "gitrevision"
+				description: *"The git revision" | string
+				default: *"master" | string
+			},...]
```

Comment:
another thing to add here would also be a namespace. I have seen multiple examples where you might want to run the pipeline (and its various tasks) in a specific namespace. So adding that here with a default would useful.

Reply:
I don't think we want to pass that as a trigger parameter, otherwise an http webhook could run code in the kube-system namespace. But we can pass it as a parameter to the cue CLI and set it on objects created in the examples.cue. Note that using a different namespace on the trigger means a new namespace for the role binding and service accounts in `platform/components/tekton/triggers/rbac.yaml`.